Mathlib/Tactic/GCongr/Core.lean

```diff
@@ -488,7 +488,7 @@ be discharged in this way, the tactic fails. -/
 syntax "rel" " [" term,* "]" : tactic
 
 elab_rules : tactic
-  | `(tactic| rel [$hyps,*]) => do
+  | `(tactic| rel [$hyps,*]) => withMainContext do
     let hyps ← hyps.getElems.mapM (elabTerm · none)
     let g ← getMainGoal
     g.withContext do
```

Comment:
Can you just reorder these lines? As long as the `elabTerm` is "inside" the `g.withContext` you should be okay. As it is you're doing the same thing twice.

Reply:
Thanks for the explanation!